pkg/kube/pod.go

```diff
@@ -133,3 +151,18 @@ func WaitForPodCompletion(ctx context.Context, cli kubernetes.Interface, namespa
 	})
 	return errors.Wrap(err, "Pod did not transition into complete state")
 }
+
+// PodSpecOverride override default pod Spec with the ones provided via specs
+func PodSpecOverride(ctx context.Context, defaultSpecs, overrideSpecs v1.PodSpec) (v1.PodSpec, error) {
```

Comment:
1. I am confused about what the semantics (expected behavior) for this function are. It is not clear from the documentation.
2. From whatever I can infer (really guess), it is not clear that the implementation below achieves that.
3. Is there test coverage for this function?